### PR TITLE
[utfcpp] Use vcpkg_*_cmake and add features

### LINKED ISSUE #13495: [utfcpp] Use vcpkg_install_cmake instead of copying include files directly
****

**Is your feature request related to a problem? Please describe.**
utfcpp does provide exported targets, which in my opinion is always better than `find_path` or `find_library`, but the current portfile for utfcpp doesn't use `vcpkg_install_cmake`, so we doesn't have exported targets to work on.

**Proposed solution**
Switch to `vcpkg_install_cmake`

### PR DESCRIPTION
**Pull request by JackBoosY**
This port provides `CMakeLists.txt`, switch to use cmake. Add feature `sample` and `test`.

Fixes #13495.

Test features successful on `x86-windows`, `x64-windows` and `x64-windows-static`.

### CODE REVIEW

**Review by NancyLi1013 — APPROVED** [review]

File: ports/utfcpp/CONTROL [hunk 1]
```diff
@@ -1,4 +1,12 @@
 Source: utfcpp
 Version: 3.1.1
+Port-Version: 1
 Homepage: https://github.com/nemtrif/utfcpp
 Description: UTF-8 with C++ in a Portable Way
+
+Feature: sample
```
Comment (ras0219-msft) [inline ports/utfcpp/CONTROL:7]: These are not good features because library consumers should not depend upon them.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Please remove the added features from this PR

File: ports/utfcpp/CONTROL [hunk 2]
```diff
@@ -1,4 +1,9 @@
 Source: utfcpp
 Version: 3.1.1
+Port-Version: 1
 Homepage: https://github.com/nemtrif/utfcpp
 Description: UTF-8 with C++ in a Portable Way
+
+Feature: test
```
Comment (ras0219-msft) [inline ports/utfcpp/CONTROL:7]: Please remove this added feature from the PR

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

**Comment by BillyONeal** [comment]
CI failures look legitimate.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**